packages/pay/src/open-payments.ts

```diff
@@ -219,46 +247,64 @@ const validateUInt64 = (o: any): Int | undefined => {
 
 const isNonNullObject = (o: any): o is Record<string, any> => typeof o === 'object' && o !== null
 
-/** Transform the Open Payments server reponse into a validated invoice */
-const validateOpenPaymentsInvoice = (o: any, queryUrl: string): Invoice | undefined => {
+/** Transform the Open Payments server reponse into a validated incoming payment */
+const validateOpenPaymentsIncomingPayment = (
+  o: any,
+  queryUrl: string
+): IncomingPayment | undefined => {
   if (!isNonNullObject(o)) {
     return
   }
 
-  const { expiresAt: expiresAtIso, account, description, amount, received } = o
+  const {
+    accountId,
+    state,
+    incomingAmount,
+    receivedAmount,
+    expiresAt: expiresAtIso,
+    description,
+    externalRef,
+    receiptsEnabled,
+  } = o
   const expiresAt = Date.parse(expiresAtIso) // `NaN` if date is invalid
-  const amountToDeliver = validateUInt64(amount)
-  const amountDelivered = validateUInt64(received)
-  const asset = validateOpenPaymentsAsset(o)
+  const amountToDeliver = validateUInt64(incomingAmount.amount)
+  const amountDelivered = validateUInt64(receivedAmount.amount)
+  const assetToDeliver = validateOpenPaymentsAsset(incomingAmount)
+  const assetDelivered = validateOpenPaymentsAsset(receivedAmount)
 
   if (
-    typeof account !== 'string' ||
+    typeof accountId !== 'string' ||
     typeof description !== 'string' ||
+    typeof externalRef !== 'string' ||
+    typeof receiptsEnabled !== 'boolean' ||
     !isNonNegativeRational(expiresAt) ||
     !amountToDeliver ||
     !amountToDeliver.isPositive() ||
     !amountDelivered ||
-    !asset
+    !assetToDeliver ||
+    !assetDelivered
   ) {
     return
   }
 
-  const accountUrl = AccountUrl.fromUrl(account)
+  const accountUrl = AccountUrl.fromUrl(accountId)
+  if (!accountUrl) return
 
-  if (!accountUrl) {
-    return
-  }
+  if (state! in IncomingPaymentState) return
```

Comment:
- Is this condition inverted?
- I can't find where the `state`'s values are specified, but I thought I remembered them being lower-cased. But `IncomingPaymentState`'s (keys and values) are upper-cased.
- Does this return `true` for states like `"toString"`?

I would rewrite this to an explicit switch-case that maps from the string `state` to a `IncomingPaymentState` enum, with a `default` branch that returns.

Reply:
I don't know what I did here. Should be fixed now. I also added a test.